quantus/helpers/model/pytorch_model.py

```diff
@@ -97,6 +98,19 @@ def _get_model_with_linear_top(self) -> torch.nn:
 
         return linear_model
 
+    def _obtain_predictions(self, x, model_predict_kwargs):
+        pred = None
+        if isinstance(self.model, PreTrainedModel):
+            if not isinstance(x, BatchEncoding):
+                raise ValueError(
+                    "When using HuggingFace pretrained models, please use Tokenizers output for `x`"
+                )
+            pred = self.model(**x, **model_predict_kwargs).logits
```

Comment:
If we do this, dont it increase the number of returns redundantly? say currently we have:
```
          pred = self.model(**x, **model_predict_kwargs).logits
          if self.softmax:
              pred = torch.softmax(pred, dim=-1)
      elif isinstance(self.model, nn.Module):
          pred_model = self.get_softmax_arg_model()
          pred = pred_model(torch.Tensor(x).to(self.device), **model_predict_kwargs)
      return pred
```

it would then become:
```
          if self.softmax:
              return torch.softmax(self.model(**x, **model_predict_kwargs).logits
, dim=-1)
      return self.model(**x, **model_predict_kwargs).logits
      elif isinstance(self.model, nn.Module):
          pred_model = self.get_softmax_arg_model()
         return pred_model(torch.Tensor(x).to(self.device), **model_predict_kwargs)
```

what do you think?

Reply:
I did slightly different to handle the softmax param and raise it properly. Could you see if you agree? Thanks

Reply:
I think that looks great :D @abarbosa94

Reply:
you're right, it looks a bit different now. Can we also remove `pred = None` at the top?